packages/aws-cdk-lib/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -1390,6 +1415,7 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
           spotOptions: props.spotPrice !== undefined ? { maxPrice: parseFloat(props.spotPrice) } : undefined,
           blockDevices: props.blockDevices,
           instanceProfile,
+          ...keyUsageType,
```

Comment:
```suggestion
          keyPair: props.keyPair,
          keyName: props.keyName,
```

> The idea of ​​creating the keyUsageType was to send just one of them (keyPair or keyName) to the LaunchTemplate, to avoid sending the keyName if keyPair is set

Makes sense, thanks for clarifying!
However, we could simply specify it like this then if I'm not mistaken.

Reply:
I tried the way you mentioned, but I believe that because of the `LaunchTemplate`, when sending the keyName in your constructor, it will display the warning on the screen, even if the user did not define it when creating the ASG :face_with_spiral_eyes: 

> [WARNING] aws-cdk-lib.aws_ec2.LaunchTemplateProps#keyName is deprecated.

Reply:
Well, that's annoying 🫠
```suggestion
          keyPair: props.keyPair,
          ...{ props.keyName ? { keyName: props.keyName } : {} },
```
I guess this should work then.
Your implementation is correct of course, but I'd like to keep it more concise if possible.
Thanks for following up on this 👍